src/models/CDD.java

```diff
@@ -11,87 +12,150 @@
+     * This property is <code>true</code> if this {@link CDD} is urgent. Otherwise, it is not urgent.
+     *
+     * If {@link DeferredProperty#get()} returns <code>null</code> then it is likely that the {@link DeferredProperty}
+     * needs to be re-computed which is the case if {@link DeferredProperty#isDirty()} returns <code>true</code>.
      */
-    private boolean isTerminal;
-    private boolean isTerminalDirty;
+    private final DeferredProperty<Boolean> isUrgentProperty = new DeferredProperty<>();
 
     /**
-     * If {@link CDD#isGuardDirty} is true then this {@link CDD} has changed, and {@link CDD#isUrgent()} will be recompute.
-     * Otherwise, {@link CDD#isUrgent()} will just return {@link CDD#isUrgent}.
+     * This property is <code>true</code> if this {@link CDD} is unrestrained (I.e., {@link CDD#equivTrue()} is <code>true</code>).
+     * Otherwise, it is not unrestrained.
+     *
+     * If {@link DeferredProperty#get()} returns <code>null</code> then it is likely that the {@link DeferredProperty}
+     * needs to be re-computed which is the case if {@link DeferredProperty#isDirty()} returns <code>true</code>.
      */
-    private boolean isUrgent;
-    private boolean isUrgentDirty;
+    private final DeferredProperty<Boolean> isUnrestrainedProperty = new DeferredProperty<>();
 
-    private boolean isUnrestrained;
-    private boolean isUnrestrainedDirty;
+    /**
+     * This property is <code>true</code> if this {@link CDD} is equivalent to the {@link CDD#cddTrue()}.
+     * However, if this {@link CDD} is not equivalent to {@link CDD#cddTrue()} then it is not guaranteed that it
+     * is equivalent to {@link CDD#cddFalse()}.
+     *
+     * If {@link DeferredProperty#get()} returns <code>null</code> then it is likely that the {@link DeferredProperty}
+     * needs to be re-computed which is the case if {@link DeferredProperty#isDirty()} returns <code>true</code>.
+     */
+    private final DeferredProperty<Boolean> isEquivTrueProperty = new DeferredProperty<>();
 
-    private boolean isTrue;
-    private boolean isTrueDirty;
+    /**
+     * This property is <code>true</code> if this {@link CDD} is equivalent to the {@link CDD#cddFalse()}.
+     * However, if this {@link CDD} is not equivalent to {@link CDD#cddFalse()} then it is not guaranteed that it
+     * is equivalent to {@link CDD#cddTrue()}.
+     *
+     * If {@link DeferredProperty#get()} returns <code>null</code> then it is likely that the {@link DeferredProperty}
+     * needs to be re-computed which is the case if {@link DeferredProperty#isDirty()} returns <code>true</code>.
+     */
+    private final DeferredProperty<Boolean> isEquivFalseProperty = new DeferredProperty<>();
 
-    private boolean isFalse;
-    private boolean isFalseDirty;
+    /**
+     * This property is <code>true</code> if this {@link CDD} can indefinitely be delayed.
+     * Otherwise, if <code>false</code> it cannot be indefinitely delayed.
```

Comment:
```suggestion
     * This property is {@code true} if this {@link CDD} can indefinitely be delayed.
     * Otherwise, if {@code false} it cannot be indefinitely delayed.
```